### PR TITLE
Start Documenting Save Data

### PR DESCRIPTION
**Pull request by ravepossum**
This is a handful of documentation for save data, including:

- Naming the savedata .c and .h files
- Naming several savedata functions/struct members/variables
- Moving the save data structs to the header file and removing the struct declaration file for SaveData
- Creating a constants file for savedata.h and save_table.h

Discord contact: ravepossum

### CODE REVIEW

File: include/constants/heap.h [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #ifndef POKEPLATINUM_CONSTANTS_HEAP_H
 #define POKEPLATINUM_CONSTANTS_HEAP_H
 
+enum {
+    HEAP_BASE_ID_SYSTEM,
+    HEAP_BASE_ID_SAVE,
+    HEAP_BASE_ID_DEBUG,
+    HEAP_BASE_ID_APPLICATION,
+    HEAP_BASE_ID_COUNT,
+};
```
Comment (lhearachel) [inline include/constants/heap.h:10]: Should these also have a defining label?

Additionally, some conventions:
1. Always explicitly define the initial value.
2. Is `COUNT` meant to be the exclusive upper bound when looping over these values? e.g.:

```c
for (int i = 0; i < HEAP_BASE_ID_COUNT; i++) { /* stuff */ }
```

If so, our convention is that such values should be labeled `MAX`, as it represents a bound and not a total count of values.

File: include/savedata.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#ifndef POKEPLATINUM_SAVEDATA_H
+#define POKEPLATINUM_SAVEDATA_H
+
+#include "constants/savedata/savedata.h"
+#include "constants/savedata/save_table.h"
+
+typedef struct {
```
Comment (lhearachel) [inline include/savedata.h:7]: Convention in the project is to specify a name for structs such that a user can choose whether they want to use `struct StructName` or `StructName` as a type-definition. The names can be identical (and, honestly, should be).

File: include/savedata.h [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    u32 unk_00;
+    u32 unk_04;
+    u32 unk_08;
+    u16 unk_0C;
+    u16 unk_0E;
+} UnkStruct_020253B4;
+
+typedef struct SaveData SaveData;
```
Comment (lhearachel) [inline include/savedata.h:79]: Superfluous typedef

File: include/savedata.h [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+SaveData* InitializeSaveData(void);
```
Comment (lhearachel) [inline include/savedata.h:81]: `SaveData_Init`

File: src/overlay005/ov5_021E1D20.c [hunk 5]
```diff
@@ -8,7 +8,7 @@
 #include "struct_decls/struct_02025E5C_decl.h"
 #include "trainer_info.h"
 #include "struct_decls/pokedexdata_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay005/ov5_021E1D20.c:11]: Duplicate include

File: src/overlay059/ov59_021D2B44.c [hunk 6]
```diff
@@ -2,7 +2,7 @@
 #include <string.h>
 
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay059/ov59_021D2B44.c:5]: Duplicate include

File: src/overlay057/ov57_021D0F30.c [hunk 7]
```diff
@@ -2,9 +2,9 @@
 #include <string.h>
 
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay057/ov57_021D0F30.c:5]: Duplicate include

File: src/overlay061/ov61_0222AE60.c [hunk 8]
```diff
@@ -4,7 +4,7 @@
 #include "strbuf.h"
 #include "struct_decls/struct_02029C68_decl.h"
 #include "struct_decls/struct_020797DC_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay061/ov61_0222AE60.c:7]: Duplicate include

File: src/overlay062/ov62_02248408.c [hunk 9]
```diff
@@ -1,7 +1,7 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay062/ov62_02248408.c:4]: Duplicate include

File: src/overlay097/ov97_0222AE60.c [hunk 10]
```diff
@@ -13,7 +13,7 @@
 #include "struct_decls/struct_02025E5C_decl.h"
 #include "trainer_info.h"
 #include "struct_decls/pokedexdata_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_0222AE60.c:16]: Duplicate include

File: src/overlay097/ov97_02230410.c [hunk 11]
```diff
@@ -18,7 +18,7 @@
 #include "strbuf.h"
 #include "struct_decls/struct_0202442C_decl.h"
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_02230410.c:21]: Duplicate include

File: src/overlay097/ov97_02232054.c [hunk 12]
```diff
@@ -3,7 +3,7 @@
 
 #include "struct_decls/struct_02018340_decl.h"
 #include "struct_decls/struct_0202442C_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_02232054.c:6]: Duplicate include

File: src/overlay097/ov97_02233B18.c [hunk 13]
```diff
@@ -16,7 +16,7 @@
 #include "strbuf.h"
 #include "struct_decls/struct_02024440_decl.h"
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_02233B18.c:19]: Duplicate include

File: src/overlay097/ov97_02237694.c [hunk 14]
```diff
@@ -12,7 +12,7 @@
 #include "struct_decls/struct_02022550_decl.h"
 #include "strbuf.h"
 #include "pokemon.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_02237694.c:15]: Duplicate include

File: src/party.c [hunk 15]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/party.c:4]: Duplicate include

File: src/poffin.c [hunk 16]
```diff
@@ -1,14 +1,14 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/poffin.c:4]: Duplicate include

File: src/party.c [hunk 17]
```diff
@@ -148,6 +148,6 @@ Party * Party_GetFromSavedata (SaveData * param0)
 {
     Party * v0;
 
-    v0 = (Party *)SaveData_Get(param0, 2);
+    v0 = (Party *)SaveData_SaveTable(param0, 2);
```
Comment (lhearachel) [inline src/party.c:151]: Does the following match?

```suggestion
    return SaveData_SaveTable(param0, 2);
```

File: src/poketch_data.c [hunk 18]
```diff
@@ -2,9 +2,9 @@
 #include <string.h>
 
 #include "pokemon.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/poketch_data.c:5]: Duplicate include

File: src/unk_02014D38.c [hunk 19]
```diff
@@ -1,7 +1,7 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02014D38.c:4]: Duplicate include

File: src/unk_0202440C.c [hunk 20]
```diff
@@ -7,14 +7,14 @@
 #include "struct_decls/struct_0202F264_decl.h"
 #include "struct_decls/struct_020308A0_decl.h"
 #include "struct_decls/struct_020797DC_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202440C.c:10]: Duplicate include

File: src/unk_02025CB0.c [hunk 21]
```diff
@@ -1,11 +1,11 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02025CB0.c:4]: Duplicate include

File: src/unk_02025E08.c [hunk 22]
```diff
@@ -2,12 +2,12 @@
 #include <string.h>
 
 #include "struct_decls/struct_02025E0C_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02025E08.c:5]: Duplicate include

File: src/unk_0202602C.c [hunk 23]
```diff
@@ -3,14 +3,14 @@
 
 #include "strbuf.h"
 #include "struct_decls/struct_02026030_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202602C.c:6]: Duplicate include

File: src/unk_02026150.c [hunk 24]
```diff
@@ -2,14 +2,14 @@
 #include <string.h>
 
 #include "struct_decls/struct_02026030_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02026150.c:5]: Duplicate include

File: src/unk_020261E4.c [hunk 25]
```diff
@@ -1,12 +1,12 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020261E4.c:4]: Duplicate include

File: src/unk_0202631C.c [hunk 26]
```diff
@@ -4,10 +4,10 @@
 #include "inlines.h"
 
 #include "pokemon.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202631C.c:7]: Duplicate include

File: src/unk_020277A4.c [hunk 27]
```diff
@@ -2,7 +2,7 @@
 #include <string.h>
 
 #include "strbuf.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020277A4.c:5]: Duplicate include

File: src/unk_02027F84.c [hunk 28]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02027F84.c:4]: Duplicate include

File: src/unk_02028124.c [hunk 29]
```diff
@@ -2,7 +2,7 @@
 #include <string.h>
 
 #include "struct_decls/struct_party_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02028124.c:5]: Duplicate include

File: src/unk_0202854C.c [hunk 30]
```diff
@@ -4,7 +4,7 @@
 #include "core_sys.h"
 
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202854C.c:7]: Duplicate include

File: src/unk_020298BC.c [hunk 31]
```diff
@@ -5,7 +5,7 @@
 
 #include "strbuf.h"
 #include "pokemon.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020298BC.c:8]: Duplicate include

File: src/unk_0202ACE0.c [hunk 32]
```diff
@@ -5,14 +5,14 @@
 #include "assert.h"
 
 #include "strbuf.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202ACE0.c:8]: Duplicate include

File: src/unk_0202B37C.c [hunk 33]
```diff
@@ -2,15 +2,15 @@
 #include <string.h>
 
 #include "strbuf.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202B37C.c:5]: Duplicate include

File: src/unk_0202B604.c [hunk 34]
```diff
@@ -3,7 +3,7 @@
 
 #include "struct_decls/struct_02025E5C_decl.h"
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202B604.c:6]: Duplicate include

File: src/unk_0202C7FC.c [hunk 35]
```diff
@@ -1,9 +1,9 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202C7FC.c:4]: Duplicate include

File: src/unk_0202C858.c [hunk 36]
```diff
@@ -1,11 +1,11 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202C858.c:4]: Duplicate include

File: src/unk_0202C9F4.c [hunk 37]
```diff
@@ -1,14 +1,14 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202C9F4.c:4]: Duplicate include

File: src/unk_0202CC64.c [hunk 38]
```diff
@@ -1,11 +1,11 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202CC64.c:4]: Duplicate include

File: src/unk_0202CC64.c [hunk 39]
```diff
@@ -29,7 +29,7 @@ ChatotCry * AllocateAndInitializeChatotCryData (int param0)
 
 ChatotCry * GetChatotCryDataFromSave (SaveData * param0)
 {
-    return (ChatotCry *)SaveData_Get(param0, 22);
+    return (ChatotCry *)SaveData_SaveTable(param0, 22);
```
Comment (lhearachel) [inline src/unk_0202CC64.c:32]: Is this cast necessary?

File: src/unk_0202CD50.c [hunk 40]
```diff
@@ -2,10 +2,10 @@
 #include <string.h>
 
 #include "struct_decls/pokedexdata_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202CD50.c:5]: Duplicate include

File: src/unk_0202D05C.c [hunk 41]
```diff
@@ -4,7 +4,7 @@
 #include "inlines.h"
 
 #include "message.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202D05C.c:7]: Duplicate include

File: src/unk_0202D778.c [hunk 42]
```diff
@@ -1,9 +1,9 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202D778.c:4]: Duplicate include

File: src/unk_0202D7A8.c [hunk 43]
```diff
@@ -1,7 +1,7 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202D7A8.c:4]: Duplicate include

File: src/unk_0202DA40.c [hunk 44]
```diff
@@ -2,9 +2,9 @@
 #include <string.h>
 
 #include "pokemon.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202DA40.c:5]: Duplicate include

File: src/unk_0202DAB4.c [hunk 45]
```diff
@@ -1,14 +1,14 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202DAB4.c:4]: Duplicate include

File: src/unk_0202E840.c [hunk 46]
```diff
@@ -5,7 +5,7 @@
 #include "trainer_info.h"
 #include "struct_decls/struct_0202CD88_decl.h"
 #include "struct_decls/struct_0203068C_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202E840.c:8]: Duplicate include

File: src/unk_0202F108.c [hunk 47]
```diff
@@ -1,11 +1,11 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202F108.c:4]: Duplicate include

File: src/unk_0202F1D4.c [hunk 48]
```diff
@@ -3,7 +3,7 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202F1D4.c:6]: Duplicate include

File: src/unk_0202FF4C.c [hunk 49]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0202FF4C.c:4]: Duplicate include

File: src/unk_02030108.c [hunk 50]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030108.c:4]: Duplicate include

File: src/unk_020302D0.c [hunk 51]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020302D0.c:4]: Duplicate include

File: src/unk_02030494.c [hunk 52]
```diff
@@ -1,13 +1,13 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030494.c:4]: Duplicate include

File: src/unk_0203061C.c [hunk 53]
```diff
@@ -1,12 +1,12 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0203061C.c:4]: Duplicate include

File: src/unk_02030880.c [hunk 54]
```diff
@@ -2,12 +2,12 @@
 #include <string.h>
 
 #include "struct_decls/struct_0203068C_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030880.c:5]: Duplicate include

File: src/unk_02030A80.c [hunk 55]
```diff
@@ -6,7 +6,7 @@
 #include "trainer_info.h"
 #include "struct_decls/struct_0202783C_decl.h"
 #include "struct_decls/struct_0202C878_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030A80.c:9]: Duplicate include

File: src/unk_02030CE8.c [hunk 56]
```diff
@@ -3,14 +3,14 @@
 
 #include "trainer_info.h"
 #include "struct_decls/struct_0202C878_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030CE8.c:6]: Duplicate include

File: src/unk_02030EA4.c [hunk 57]
```diff
@@ -1,9 +1,9 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02030EA4.c:4]: Duplicate include

File: src/unk_020366A0.c [hunk 58]
```diff
@@ -6,15 +6,15 @@
 
 #include "struct_decls/struct_02018340_decl.h"
 #include "trainer_info.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020366A0.c:9]: Duplicate include

File: src/unk_02038ED4.c [hunk 59]
```diff
@@ -1,9 +1,9 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02038ED4.c:4]: Duplicate include

File: src/unk_0203A6DC.c [hunk 60]
```diff
@@ -1,15 +1,15 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0203A6DC.c:4]: Duplicate include

File: src/unk_0203D1B8.c [hunk 61]
```diff
@@ -18,7 +18,7 @@
 #include "struct_decls/struct_0207AE68_decl.h"
 #include "struct_decls/struct_0207D3C0_decl.h"
 #include "struct_decls/struct_0209747C_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0203D1B8.c:21]: Duplicate include

File: src/unk_0204CA84.c [hunk 62]
```diff
@@ -8,12 +8,12 @@
 #include "struct_decls/struct_02026310_decl.h"
 #include "struct_decls/struct_0203E724_decl.h"
 #include "struct_decls/struct_party_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0204CA84.c:11]: Duplicate include

File: src/unk_0204F13C.c [hunk 63]
```diff
@@ -12,7 +12,7 @@
 #include "struct_decls/struct_020508D4_decl.h"
 #include "pokemon.h"
 #include "struct_decls/struct_party_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0204F13C.c:15]: Duplicate include

File: src/unk_020507CC.c [hunk 64]
```diff
@@ -1,9 +1,9 @@
 #include <nitro.h>
 #include <string.h>
 
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_020507CC.c:4]: Duplicate include

File: src/unk_0206CCB0.c [hunk 65]
```diff
@@ -16,7 +16,7 @@
 #include "struct_decls/struct_020507E4_decl.h"
 #include "pokemon.h"
 #include "struct_decls/struct_party_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0206CCB0.c:19]: Duplicate include

File: src/unk_0207D3B8.c [hunk 66]
```diff
@@ -2,12 +2,12 @@
 #include <string.h>
 
 #include "struct_decls/struct_0207CB08_decl.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0207D3B8.c:5]: Duplicate include

File: src/unk_02099DFC.c [hunk 67]
```diff
@@ -8,7 +8,7 @@
 #include "message.h"
 #include "struct_decls/struct_02018340_decl.h"
 #include "strbuf.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_02099DFC.c:11]: Duplicate include

File: src/unk_0209A2C4.c [hunk 68]
```diff
@@ -8,7 +8,7 @@
 #include "message.h"
 #include "struct_decls/struct_02018340_decl.h"
 #include "strbuf.h"
-#include "struct_decls/struct_021C0794_decl.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/unk_0209A2C4.c:11]: Duplicate include

File: src/savedata.c [hunk 69]
```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
...
-    return SaveData_Get((SaveData *)param0, param1);
+    return SaveData_SaveTable((SaveData*)saveData, saveTableID);
 }
 
-BOOL sub_020245E8 (SaveData * param0)
+BOOL SaveData_Erase (SaveData *saveData)
 {
-    int v0, v1;
-    u8 * v2 = Heap_AllocFromHeapAtEnd(3, 0x1000);
+    int sectorIndex, unused;
```
Comment (lhearachel) [inline src/savedata.c:123]: Is `unused` not stripped by the compiler? Does it not match if you remove the extraneous var?

Additionally, can you move `sectorIndex` further down?

File: src/savedata.c [hunk 70]
```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
...
+        saveData->dataExists = TRUE;
+        saveData->isNewGameData = FALSE;
 
         {
-            int v1, v2;
+            int frontierResult, videoResult;
 
-            sub_02024CD4(param0, &v1, &v2);
+            SaveDataExtra_LoadCheck(saveData, &frontierResult, &videoResult);
         }
```
Comment (lhearachel) [inline src/savedata.c:167]: Unnecessary scope-block

File: src/savedata.c [hunk 71]
```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
...
 
-        return 1;
+        return TRUE;
     } else {
-        return 0;
+        return FALSE;
     }
```
Comment (lhearachel) [inline src/savedata.c:172]: Does this have to exist in an `else` block?

File: src/savedata.c [hunk 72]
```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
...
 }
 
-int sub_020246E0 (SaveData * param0)
+int SaveData_Save (SaveData *saveData)
 {
-    int v0;
+    int saveResult;
```
Comment (lhearachel) [inline src/savedata.c:177]: If you can declare this alongside the definition, do so; we use C99-style for this project.

I won't point out the other instances in the file, just making a note of this one.

File: src/savedata.c [hunk 73]
```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
...
+        sub_0202516C(saveData, SAVE_BLOCK_ID_NORMAL, !saveData->unk_20020[SAVE_BLOCK_ID_NORMAL]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, !saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_NORMAL, saveData->unk_20020[SAVE_BLOCK_ID_NORMAL]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
 
         SleepUnlock(1);
     }
 
-    v0 = sub_0202513C(param0);
+    saveResult = sub_0202513C(saveData);
```
Comment (lhearachel) [inline src/savedata.c:194]: Continuation from the above:

```suggestion
    int saveResult = sub_0202513C(saveData);
```

**Review by lhearachel — CHANGES_REQUESTED** [review]
Apologies in advance for the noise of "duplicate includes"; if you want to modify a previous commit and do a force-push to make that easier, feel free to do so.

File: include/constants/heap.h (hunk 1 above)
Comment (cbt6) [inline include/constants/heap.h:10]:
```suggestion
enum HeapId {
    HEAP_ID_SYSTEM = 0,
    HEAP_ID_SAVE,
    HEAP_ID_DEBUG,
    HEAP_ID_APPLICATION,

    HEAP_ID_BATTLE = 5,
};
```

File: include/constants/savedata/save_table.h [hunk 74]
```diff
@@ -0,0 +1,70 @@
...
+
+    EXTRA_SAVE_TABLE_ENTRY_MAX
+};
+
+#define SAVE_PAGE_MAX       32
+#define SAVE_SECTOR_SIZE    (0x1000)
+
+#define EXTRA_SAVE_TABLE_ENTRY_NONE 0xFFFFFFFF
+
+#endif // POKEPLATINUM_CONSTANTS_SAVE_TABLE_H
```
Comment (cbt6) [inline include/constants/savedata/save_table.h:70]: Add newline to EOF (and for all other new files)

**Comment by ravepossum** [comment]
All the feedback from those reviews should be addressed now.

**Review by lhearachel — APPROVED** [review]

**Pull request merged**